iaso/tasks/process_mobile_bulk_upload.py

```diff
@@ -96,19 +98,67 @@ def process_mobile_bulk_upload(api_import_id, project_id, task=None):
     )
 
 
-def process_instance_xml(directory, zip_ref, user):
-    instance = Instance.objects.get(uuid=directory.name)
+def read_json_file_from_zip(zip_ref, filename):
+    with zip_ref.open(filename) as file:
+        return json.load(file)
+
+
+def get_directory_handlers(zip_ref):
+    result = {}
+    for directory in zipfile.Path(zip_ref).iterdir():
+        if directory.is_dir():
+            result[directory.name] = directory
+    return result
+
+
+def process_instance_xml(uuid, instance_data, zip_ref, user):
+    instance = Instance.objects.get(uuid=uuid)
```

Comment:
It seems that `uuid` can be `None`, so this `.get()` has the potential to fail.

Should we be more defensive and catch the potential error?

Reply:
Good catch! Now that I re-read it, `uuid` should not be `None` in the `instances.json` file. So I'm updating the `uuid = instance_data.get("id", None)`, it should fail if the id is None IMO.